Intermittent Bag generation failure

QDR noticed an intermittent issue with Bag creation that is caused by a NoSuchElementException thrown from within the Apache CloseableHttpClient class: 
```
Caused by: java.util.NoSuchElementException
        at java.util.AbstractList$Itr.next(AbstractList.java:364)
        at java.util.AbstractList.removeRange(AbstractList.java:570)
        at java.util.AbstractList.clear(AbstractList.java:234)
        at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:103)
        at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
        at edu.harvard.iq.dataverse.util.bagit.BagGenerator$3.get(BagGenerator.java:1002)
```
Investigating that it appears that reusing an HttpClientContext across threads is a problem (sounds like while the HttpClientContext class itself is threadsafe, some of the classes that it uses are not.) Since that class was only included to manage cookies (in other applications - Dataverse's API doesn't set a session cookie), a fix is to just remove it. PR to follow.

--- a/src/main/java/edu/harvard/iq/dataverse/util/bagit/BagGenerator.java
+++ b/src/main/java/edu/harvard/iq/dataverse/util/bagit/BagGenerator.java
@@ -93,7 +93,6 @@ public class BagGenerator {
     private int timeout = 60;
     private RequestConfig config = RequestConfig.custom().setConnectTimeout(timeout * 1000)
             .setConnectionRequestTimeout(timeout * 1000).setSocketTimeout(timeout * 1000).build();
-    private static HttpClientContext localContext = HttpClientContext.create();
     protected CloseableHttpClient client;
     private PoolingHttpClientConnectionManager cm = null;
 
@@ -986,7 +985,8 @@ public class BagGenerator {
                         HttpGet getMap = createNewGetRequest(new URI(uri), null);
                         logger.finest("Retrieving " + tries + ": " + uri);
                         CloseableHttpResponse response;
-                        response = client.execute(getMap, localContext);
+                        //Note - if we ever need to pass an HttpClientContext, we need a new one per thread.
+                        response = client.execute(getMap);
                         if (response.getStatusLine().getStatusCode() == 200) {
                             logger.finest("Retrieved: " + uri);
                             return response.getEntity().getContent();

